selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/git/GitPullConflictTest.java

```diff
@@ -0,0 +1,199 @@
+import static org.eclipse.che.selenium.core.constant.TestMenuCommandsConstants.Git.Remotes.REMOTES_TOP;
+import static org.eclipse.che.selenium.core.constant.TestTimeoutsConstants.LOADER_TIMEOUT_SEC;
+import static org.eclipse.che.selenium.pageobject.Wizard.TypeProject.MAVEN;
+
+import com.google.inject.Inject;
+import com.google.inject.name.Named;
+import java.io.IOException;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import org.eclipse.che.commons.lang.NameGenerator;
+import org.eclipse.che.selenium.core.client.TestGitHubServiceClient;
+import org.eclipse.che.selenium.core.client.TestSshServiceClient;
+import org.eclipse.che.selenium.core.client.TestUserPreferencesServiceClient;
+import org.eclipse.che.selenium.core.constant.TestGitConstants;
+import org.eclipse.che.selenium.core.constant.TestMenuCommandsConstants;
+import org.eclipse.che.selenium.core.user.TestUser;
+import org.eclipse.che.selenium.core.workspace.TestWorkspace;
+import org.eclipse.che.selenium.pageobject.CodenvyEditor;
+import org.eclipse.che.selenium.pageobject.Consoles;
+import org.eclipse.che.selenium.pageobject.Events;
+import org.eclipse.che.selenium.pageobject.Ide;
+import org.eclipse.che.selenium.pageobject.Loader;
+import org.eclipse.che.selenium.pageobject.Menu;
+import org.eclipse.che.selenium.pageobject.ProjectExplorer;
+import org.eclipse.che.selenium.pageobject.git.Git;
+import org.kohsuke.github.GHRepository;
+import org.kohsuke.github.GitHub;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+/** @author aleksandr shmaraev */
+public class GitPullConflictTest {
+  private static final String REPO_NAME = NameGenerator.generate("PullConflictTest-", 3);
+  private static final String PROJECT_NAME = NameGenerator.generate("PullConflictProject-", 4);
+  private static final String pathToJavaFile = "src/main/java/commenttest";
+  private static final String COMMIT_MSG = "commit_changes";
+
+  private static final String firstMergeConflictMessage =
+      "Checkout operation failed, the following files would be overwritten by merge:\n"
+          + "GitPullTest.txt\n"
+          + "src/main/java/commenttest/JavaCommentsTest.java\n"
+          + "Could not pull. Commit your changes before merging.";
+
+  private static final String secondMergeConflictMessage =
+      "Could not pull because a merge conflict is detected in the files:\n"
+          + "GitPullTest.txt\n"
+          + "src/main/java/commenttest/JavaCommentsTest.java\n"
+          + "Automatic merge failed; fix conflicts and then commit the result.";
+
+  private static final String CHANGE_STRING_1 =
+      String.format("//first_change_%s", System.currentTimeMillis());
+
+  private static final String headConfPrefixConfMess =
+      "<<<<<<< HEAD\n" + "//second_change\n" + "=======\n" + CHANGE_STRING_1 + "\n" + ">>>>>>>";
```

Comment:
Too many concatenations makes makes it hard to read. Please, replace concatenations with one `String.format()` method invocation.